### PR TITLE
fix:(@desktop/Profile): Preview Bio length and scroll area fix #7922

### LINKED ISSUE #7922: ProfileDialog: bio text should have maximum height defined
# Bug Report

## Description

The dialog doesn't look correct when bio has many new lines.

[Kazam_screencast_00063.webm](https://user-images.githubusercontent.com/20650004/196141705-1644d747-ea99-4416-8e65-dc2315ae816f.webm)

Additionally profile edit form doesn't behave properly when vertically long bio is set:

[Kazam_screencast_00064.webm](https://user-images.githubusercontent.com/20650004/196141723-386402ab-f4a6-490f-8fb3-31a69cfe9b32.webm)

## Steps to reproduce

Set bio with many new lines

#### Expected behavior

Bio text should have maximum height, when more space is required it should be scrollable.

#### Actual behavior

the whole modal content is very long

### PR DESCRIPTION
**Pull request by mprakhov**
### What does the PR do

Preview Bio length and scroll area fix #7922
- fix incorrect scroll area size if set a lot of lines in Bio
- added scroll area for Bio with a big amount of lines 

### Affected areas

Preview Bio widget

### Video:

https://user-images.githubusercontent.com/117639195/201348172-af99f221-b8a5-446f-841d-b9a3167f5dd8.mov

### CODE REVIEW

**Comment by PascalPrecht** [comment]
Can you please add a video/screenshot to this PR that shows that it/what works

File: ui/app/AppLayouts/Profile/views/MyProfileView.qml [hunk 1]
```diff
@@ -66,6 +66,24 @@ SettingsContentBase {
...
+            onImplicitHeightChanged: {
+                // During opening Profile, StackLayout always set implicitHeight of his biggest
+                // child implicitHeight and it reflecting on a scroll area.
+                // Use implicitHeight of current child
+                if (currentIndex === 0 && implicitHeight !== settingsView.implicitHeight) {
+                    implicitHeight = settingsView.implicitHeight
+                } else if (currentIndex === 1 && implicitHeight !== profilePreview.implicitHeight) {
+                    implicitHeight = profilePreview.implicitHeight
+                }
+            }
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/views/MyProfileView.qml:78]: In general, I don't like this imperative approach; also hardcoding "currentIndex" to some magic values

File: ui/app/AppLayouts/Profile/views/MyProfileView.qml [hunk 2]
```diff
@@ -66,6 +66,24 @@ SettingsContentBase {
...
+
+            onCurrentIndexChanged: {
+                switch(currentIndex) {
+                case 0: implicitHeight = settingsView.implicitHeight; break
+                case 1: implicitHeight = profilePreview.implicitHeight; break
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/views/MyProfileView.qml:83]: Implicit values should never be assigned to

File: ui/imports/shared/panels/ProfileBioSocialsPanel.qml [hunk 3]
```diff
@@ -84,13 +85,27 @@ Control {
     contentItem: ColumnLayout {
         spacing: 20
 
-        StatusBaseText {
+        StatusScrollView {
+            id: scrollView
+            visible: root.bio !== ""
```
Comment (borismelnik) [inline ui/imports/shared/panels/ProfileBioSocialsPanel.qml:90]: small suggestion: `visible: root.bio` is valid condition for this :)

File: ui/app/AppLayouts/Profile/views/MyProfileView.qml (hunk 1 above)
Comment (borismelnik) [reply ui/app/AppLayouts/Profile/views/MyProfileView.qml:78]: Good idea to give some names for this values :)

File: ui/app/AppLayouts/Profile/views/MyProfileView.qml (hunk 2 above)
Comment (mprakhov) [reply ui/app/AppLayouts/Profile/views/MyProfileView.qml:83]: Do I understand correctly that you suggest using `height` instead?

Comment (mprakhov) [reply ui/app/AppLayouts/Profile/views/MyProfileView.qml:83]: Can I disagree with you?
Implicit size is supposed to be used when calculating the size of an item based on its contents. Height on a parent item may affect the size of its children and it should never be the case when you set implicit size.

File: ui/app/AppLayouts/Profile/views/MyProfileView.qml (hunk 1 above)
Comment (mprakhov) [reply ui/app/AppLayouts/Profile/views/MyProfileView.qml:78]: fixed

File: ui/imports/shared/panels/ProfileBioSocialsPanel.qml (hunk 3 above)
Comment (mprakhov) [reply ui/imports/shared/panels/ProfileBioSocialsPanel.qml]: fixed

File: ui/imports/shared/panels/ProfileBioSocialsPanel.qml [hunk 4]
```diff
@@ -84,13 +85,27 @@ Control {
...
             Layout.fillWidth: true
-            visible: text !== ""
-            text: root.bio
-            wrapMode: Text.Wrap
-            font.weight: Font.Medium
-            lineHeight: 1.2
+            contentHeight: bioText.height
+            ScrollBar.vertical: StatusScrollBar {
+                id: scrollBar
+                policy: bioText.height > scrollView.availableHeight ? ScrollBar.AlwaysOn : ScrollBar.AlwaysOff
```
Comment (MishkaRogachev) [inline ui/imports/shared/panels/ProfileBioSocialsPanel.qml:98]: It's supposed to be working out of the box with `ScrollBar.AsNeeded`. If its not working is component bug (probably mine):
https://github.com/status-im/status-desktop/blob/master/ui/StatusQ/src/StatusQ/Core/StatusScrollView.qml#L65
https://github.com/status-im/status-desktop/blob/master/ui/StatusQ/src/StatusQ/Controls/StatusScrollBar.qml#L30

Comment (mprakhov) [reply ui/imports/shared/panels/ProfileBioSocialsPanel.qml:98]: The problem is that scrollbar is not visible if content requires scrolling until you start scrolling.
After scrolling stops - the scrollbar disappears. I force it to be visible, if we have scrollable content, in order user understood, that there are more lines of Bio

Comment (MishkaRogachev) [reply ui/imports/shared/panels/ProfileBioSocialsPanel.qml:98]: Got it, no questions :)

File: ui/app/AppLayouts/Profile/views/MyProfileView.qml (hunk 2 above)
Comment (stefandunca) [reply ui/app/AppLayouts/Profile/views/MyProfileView.qml]: Agree with both. As I see it, the `implicit*` sizes are hints for clients on the content size of a resizable control and should be set from the component's implementation. Implicit sizes are taken into consideration as hints for layouting if `Layout` attached property hints are missing. In this case `Layout.prefferedWidth` and `Layout.prefferedHeight` have the same effect and show better the intention.

Comment (osmaczko) [reply ui/app/AppLayouts/Profile/views/MyProfileView.qml]:
```diff
diff --git a/ui/app/AppLayouts/Profile/views/MyProfileView.qml b/ui/app/AppLayouts/Profile/views/MyProfileView.qml
index e99f6581e..ef2f33575 100644
--- a/ui/app/AppLayouts/Profile/views/MyProfileView.qml
+++ b/ui/app/AppLayouts/Profile/views/MyProfileView.qml
@@ -63,24 +63,28 @@ SettingsContentBase {
         width: root.contentWidth
 
         StackLayout {
+            id: stackLayout
             Layout.fillWidth: true
             currentIndex: editPreviwTabBar.currentIndex
 
             MyProfileSettingsView {
                 id: settingsView
-                Layout.fillWidth: true
                 profileStore: root.profileStore
                 privacyStore: root.privacyStore
                 walletStore: root.walletStore
+
+                onVisibleChanged: if (visible) stackLayout.Layout.preferredHeight = Qt.binding(() => settingsView.implicitHeight)
+                Component.onCompleted: stackLayout.Layout.preferredHeight = Qt.binding(() => settingsView.implicitHeight)
             }
 
             MyProfilePreview {
                 id: profilePreview
-                Layout.fillWidth: true
                 profileStore: root.profileStore
                 contactsStore: root.contactsStore
                 dirtyValues: settingsView.dirtyValues
                 dirty: settingsView.dirty
+
+                onVisibleChanged: if (visible) stackLayout.Layout.preferredHeight = Qt.binding(() => profilePreview.implicitHeight)
             }
         }
```

@mprakhov what do you think about doing it in a more declarative way?

Comment (mprakhov) [reply ui/app/AppLayouts/Profile/views/MyProfileView.qml]: this one looks really great. Refactored 👍

**Review by MishkaRogachev — APPROVED** [review]
Awesome!

**Review by caybro — APPROVED** [review]

**Review by elina2015 — APPROVED** [review]
Tested and approved

**Pull request merged**